bin/runtime-common/src/messages.rs

```diff
@@ -0,0 +1,539 @@
+		const INSTANCE: InstanceId = *b"brdg";
+		const RELAYER_FEE_PERCENT: u32 = 20;
+
+		type ThisChain = BridgedChain;
+		type BridgedChain = ThisChain;
+
+		fn maximal_dispatch_weight_of_message_on_bridged_chain() -> Weight {
+			unreachable!()
+		}
+
+		fn weight_of_delivery_transaction() -> Weight {
+			unreachable!()
+		}
+
+		fn weight_of_delivery_confirmation_transaction_on_this_chain() -> Weight {
+			unreachable!()
+		}
+
+		fn weight_of_reward_confirmation_transaction_on_target_chain() -> Weight {
+			unreachable!()
+		}
+
+		fn this_weight_to_balance(_weight: Weight) -> BridgedChainBalance {
+			unreachable!()
+		}
+
+		fn bridged_weight_to_balance(_weight: Weight) -> ThisChainBalance {
+			unreachable!()
+		}
+
+		fn this_chain_balance_to_bridged_chain_balance(_this_balance: BridgedChainBalance) -> ThisChainBalance {
+			unreachable!()
+		}
+	}
+
+	#[derive(Debug, PartialEq, Decode, Encode)]
+	struct ThisChainAccountId(u32);
+	#[derive(Debug, PartialEq, Decode, Encode)]
+	struct ThisChainSigner(u32);
+	#[derive(Debug, PartialEq, Decode, Encode)]
+	struct ThisChainSignature(u32);
+	#[derive(Debug, PartialEq, Decode, Encode)]
+	enum ThisChainCall {
+		#[codec(index = 42)]
+		Transfer,
+		#[codec(index = 84)]
+		Mint,
+	}
+
+	#[derive(Debug, PartialEq, Decode, Encode)]
+	struct BridgedChainAccountId(u32);
+	#[derive(Debug, PartialEq, Decode, Encode)]
+	struct BridgedChainSigner(u32);
+	#[derive(Debug, PartialEq, Decode, Encode)]
+	struct BridgedChainSignature(u32);
+	#[derive(Debug, PartialEq, Decode, Encode)]
+	enum BridgedChainCall {}
+
+	macro_rules! impl_wrapped_balance {
+		($name:ident) => {
+			#[derive(Debug, PartialEq, Decode, Encode, Clone, Copy)]
+			struct $name(u32);
+
+			impl From<u32> for $name {
+				fn from(balance: u32) -> Self {
+					Self(balance)
+				}
+			}
+
+			impl sp_std::ops::Add for $name {
+				type Output = $name;
+
+				fn add(self, other: Self) -> Self {
+					Self(self.0 + other.0)
+				}
+			}
+
+			impl sp_std::ops::Div for $name {
+				type Output = $name;
+
+				fn div(self, other: Self) -> Self {
+					Self(self.0 / other.0)
+				}
+			}
+
+			impl sp_std::ops::Mul for $name {
+				type Output = $name;
+
+				fn mul(self, other: Self) -> Self {
+					Self(self.0 * other.0)
+				}
+			}
+
+			impl sp_std::cmp::PartialOrd for $name {
+				fn partial_cmp(&self, other: &Self) -> Option<sp_std::cmp::Ordering> {
+					self.0.partial_cmp(&other.0)
+				}
+			}
+
+			impl CheckedAdd for $name {
+				fn checked_add(&self, other: &Self) -> Option<Self> {
+					self.0.checked_add(other.0).map(Self)
+				}
+			}
+
+			impl CheckedDiv for $name {
+				fn checked_div(&self, other: &Self) -> Option<Self> {
+					self.0.checked_div(other.0).map(Self)
+				}
+			}
+
+			impl CheckedMul for $name {
+				fn checked_mul(&self, other: &Self) -> Option<Self> {
+					self.0.checked_mul(other.0).map(Self)
+				}
+			}
+		};
+	}
+
+	impl_wrapped_balance!(ThisChainBalance);
+	impl_wrapped_balance!(BridgedChainBalance);
```

Comment:
To test that the module is able to work with different balance types on different chains. I have had a few typos (because of copy-paste) in this code, so I decided it would be better to be sure that we're using single balance types in expressions. The same thing should probably be done with the weight later.

Reply:
Can you make a note of it in the code? I can definitely see myself touching this code in the future and changing it to `u128` without remembering why it was done this way initially